api/src/api/userRouter.ts

```diff
@@ -60,15 +61,23 @@ const shouldCheckLicense = (userData: UserData): boolean => {
 };
 
 const shouldUpdateBusinessNameSearch = (userData: UserData): boolean => {
-  if (!userData.formationData.businessNameAvailability?.lastUpdatedTimeStamp) {
+  if (
+    !userData.formationData.businessNameAvailability?.lastUpdatedTimeStamp &&
+    !userData.formationData.dbaBusinessNameAvailability?.lastUpdatedTimeStamp
+  ) {
     return false;
   }
-  return (
-    userData.profileData.businessName !== undefined &&
-    userData.formationData.businessNameAvailability !== undefined &&
-    hasBeenMoreThanOneHour(userData.formationData.businessNameAvailability.lastUpdatedTimeStamp) &&
-    userData.formationData.completedFilingPayment !== true
-  );
+
+  const isDba = userData.profileData.needsNexusDbaName;
+  const shouldUpdateNameAvailability = isDba
+    ? userData.profileData.nexusDbaName !== undefined &&
+      userData.formationData.dbaBusinessNameAvailability !== undefined &&
+      hasBeenMoreThanOneHour(userData.formationData.dbaBusinessNameAvailability.lastUpdatedTimeStamp)
+    : userData.profileData.businessName !== undefined &&
+      userData.formationData.businessNameAvailability !== undefined &&
+      hasBeenMoreThanOneHour(userData.formationData.businessNameAvailability.lastUpdatedTimeStamp);
```

Comment:
This is a lot of &&

Maybe could be refactored and broken down into separate statements / variables to be more readable?